src/Shared/FileUtilities.cs

```diff
@@ -363,7 +363,7 @@ internal static string MaybeAdjustFilePath(string value, string baseDirectory =
 
             // Don't bother with arrays or properties or network paths, or those that
             // have no slashes.
-            if (NativeMethodsShared.IsWindows || string.IsNullOrEmpty(value)
+            if (Path.DirectorySeparatorChar == '\\' || string.IsNullOrEmpty(value)
```

Comment:
IsWindows and IsUnixLike are very cheap, but since they can be called millions of times here, the string compare they perform internally starts to add up. The alternative is locking and caching the values but not sure if it's worth it just for this method.